src/preact/utils.js

```diff
@@ -14,3 +14,17 @@ export function useResourcesNotify() {
     }
   });
 }
+
+/**
+ * Combines multiple refs to pass into `ref` prop.
+ * @param {...any} refs
+ * @return {function(!Element)}
+ */
+export function refs(...refs) {
+  return (element) => {
```

Comment:
Use `useCallback` to avoid trashing the ref.

Reply:
Cant use `useCallback` outside a component or hook.

Reply:
Make it a hook, then. As is, this is going to call each `refs` every time it renders.

Reply:
This is pretty clever: https://github.com/theKashey/use-callback-ref#usemergerefs

Reply:
Merged this PR as it has too many comments as it is. Created #36024 for the custom hook implementation.